lib/Auth/Process/SmartID.php

```diff
@@ -109,10 +120,16 @@ private function addID($attributes, $request)
         /*
          * At this stage no usable id_candidate has been detected.
          */
-        throw new \SimpleSAML\Error\Exception('This service needs at least one of the following
+	if ($this->fail_if_empty) {
+	     throw new \SimpleSAML\Error\Exception('This service needs at least one of the following
             attributes to identity users: '.implode(', ', $this->candidates).'. Unfortunately not
             one of them was detected. Please ask your institution administrator to release one of
             them, or try using another identity provider.');
+        } else {
+	    // return an empty identifier, missing id attribute must be handled
+	    // by another authproc filter
+            return ''; 
```

Comment:
Nitpick, but your spacing is off here

Reply:
My IDE is Emacs... Whatever IDE you are using, if spacing is important, could you adapt / auto-format this to your liking please?